### PR TITLE
chore(vendor): update go-common-libs/go-iscsi-utils and resolve breaking changes

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7960

Since longhorn-engine updates `go-common-libs`/`go-iscsi-helper`
We need to update longhorn-engine in longhonr-instance-manager's vendor

Dependency PR:
- `longhorn-engine`: https://github.com/longhorn/longhorn-engine/pull/1050

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -66,3 +65,5 @@ require (
 )
 
 replace golang.org/x/text v0.3.2 => golang.org/x/text v0.3.3
+
+replace github.com/longhorn/longhorn-engine v1.5.4 => github.com/ChanYiLin/longhorn-engine v1.4.0-rc1.0.20240311091244-3365ed4916a7
```
Comment (innobead) [inline go.mod:69]: This is not updated yet?

**Review by innobead — COMMENTED** [review]
If this PR is not ready, should mark as draft.

File: go.mod (hunk 1 above)
Comment (ChanYiLin) [reply go.mod:69]: Sure I can change it to Draft
I use to mention what this PR is depended on in the comment above

> Dependency PR:
longhorn-engine: https://github.com/longhorn/longhorn-engine/pull/1050

**Comment by innobead** [comment]
@ChanYiLin Let's merge this after the confit is fixed. Thanks.

**Review by innobead — APPROVED** [review]

**Pull request merged**